traits/trait_handlers.py

```diff
@@ -2757,6 +2757,7 @@ def update ( self, value ):
         if not hasattr(self, 'trait'):
             return set.update(self, value)
         try:
+            value = set(value)
```

Comment:
@cfarrow: Okay, thanks. It looks like Corran's addressed this. Are you okay with me merging this in its current state?